team-01/src/main/java/ubc/cosc322/AmazonsLocalBoard.java

```diff
@@ -59,4 +60,11 @@ private int getIndex(ArrayList<Integer> position) {
 		int col = position.get(1);
 		return TOTAL_LENGTH - row * ROW_LENGTH + col;
 	}
+
+	public AmazonsLocalBoard getDeepCopy() {
+		AmazonsLocalBoard copy = new AmazonsLocalBoard();
+		copy.localPlayer = localPlayer;
+		copy.state = state.stream().collect(Collectors.toCollection(ArrayList::new));
```

Comment:
Never seen that before, interesting.

Reply:
Yeah, I started learning about streams at Amazon. I like them because it saves me a few lines of a for loop and they're also *usually* more readable and a bit more declarative.